scripts/build_test.py

```diff
@@ -721,4 +723,169 @@ def mock_require_compiled_js_dir_to_be_valid():
                 self.assertFalse(
                     os.path.exists(os.path.dirname(MOCK_COMPILED_JS_DIR)))
 
+    def test_generate_app_yaml(self):
+        temp_file = tempfile.NamedTemporaryFile()
+        temp_file.name = 'app_dev.yaml'
+        with open('app_dev.yaml', 'w') as tmp:
+            tmp.write('Some content in app_dev.yaml')
+
+        build.generate_app_yaml()
+        with open('app.yaml', 'r') as yaml_file:
+            content = yaml_file.read()
+
+        temp_file.close()
+        checkout_cmd = 'git checkout -- app_dev.yaml'.split()
```

Comment:
Why do you need this? Tests should not be calling git commands.

Instead, a strategy you can use is as follows: in build.generate_app_yaml(), pull out app_dev.yaml and app.yaml into named constants. Also pre-populate a mock_app_dev.yaml file in core/tests/data with 'Some content in mock_app_dev.yaml'.

Then, in the test, swap the aforementioned named constants for paths in core/tests/data or for tempfiles. Verify the content of mock_app_dev.yaml is what you want it to be, and (if you aren't using a tempfile for mock_app.yaml) that mock_app.yaml is empty or doesn't exist. Then call the build method and then verify the contents of both mock_app_dev.yaml (should be unchanged) and mock_app.yaml. Finally, clean up if you created any new files.

You can also replace mock_app_dev.yaml with a tempfile but that's the basic idea. Whatever you do, you shouldn't be calling git commands or doing other complicated things that interact with the "outside world" in tests.

Reply:
Done